pkg/function/backup_data_using_kopia_server.go

```diff
@@ -0,0 +1,245 @@
+	}
+
+	ctx = field.Context(ctx, consts.PodNameKey, pod)
+	ctx = field.Context(ctx, consts.ContainerNameKey, container)
+
+	repositoryServerService, err := cli.CoreV1().Services(tp.RepositoryServer.Namespace).Get(context.Background(), tp.RepositoryServer.ServerInfo.ServiceName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.New("Unable to find Service Details for Repository Server")
+	}
+	repositoryServerServicePort := strconv.Itoa(int(repositoryServerService.Spec.Ports[0].Port))
+	serverAddress := "https://" + tp.RepositoryServer.ServerInfo.ServiceName + "." + tp.RepositoryServer.Namespace + ".svc.cluster.local:" + repositoryServerServicePort
+
+	snapInfo, err := backupDataUsingKopiaServer(
+		cli,
+		container,
+		hostname,
+		includePath,
+		namespace,
+		pod,
+		serverAddress,
+		fingerprint,
+		username,
+		userAccessPassphrase,
+		tags,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to backup data to Kopia API server")
+	}
+
+	var logSize, phySize int64
+	if snapInfo.Stats != nil {
+		stats := snapInfo.Stats
+		logSize = stats.SizeHashedB + stats.SizeCachedB
+		phySize = stats.SizeUploadedB
+	}
+
+	output := map[string]any{
+		BackupDataOutputBackupID:           snapInfo.SnapshotID,
+		BackupDataOutputBackupSize:         humanize.Bytes(uint64(logSize)),
+		BackupDataOutputBackupPhysicalSize: humanize.Bytes(uint64(phySize)),
+	}
+	return output, nil
+}
+
+func backupDataUsingKopiaServer(
+	cli kubernetes.Interface,
+	container,
+	hostname,
+	includePath,
+	namespace,
+	pod,
+	serverAddress,
+	fingerprint,
+	username,
+	userPassphrase string,
+	tags []string,
+) (info *kopiacmd.SnapshotCreateInfo, err error) {
+	contentCacheMB, metadataCacheMB := kopiacmd.GetCacheSizeSettingsForSnapshot()
+	configFile, logDirectory := kankopia.GetCustomConfigFileAndLogDirectory(hostname)
+
+	cmd := kopiacmd.RepositoryConnectServerCommand(kopiacmd.RepositoryServerCommandArgs{
+		UserPassword:    userPassphrase,
+		ConfigFilePath:  configFile,
+		LogDirectory:    logDirectory,
+		CacheDirectory:  kopiacmd.DefaultCacheDirectory,
+		Hostname:        hostname,
+		ServerURL:       serverAddress,
+		Fingerprint:     fingerprint,
+		Username:        username,
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+	})
+
+	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
+	format.Log(pod, container, stdout)
+	format.Log(pod, container, stderr)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to connect to Kopia API server")
+	}
+
+	cmd = kopiacmd.SnapshotCreate(
+		kopiacmd.SnapshotCreateCommandArgs{
+			PathToBackup: includePath,
+			CommandArgs: &kopiacmd.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: configFile,
+				LogDirectory:   logDirectory,
+			},
+			Tags:                   tags,
+			ProgressUpdateInterval: 0,
+		})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to generate snapshot create command")
```

Comment:
```suggestion
		return nil, errors.Wrap(err, "Failed to create snapshot create command")
```

Reply:
Let me know WDYT @viveksinghggits

Reply:
mmm, I suggested `create` because `generate` sounds like an automated system is generating the command. But that's not the case here, we are forming the command ourselves for the passed arguments. But let's ask for others' opinion as well.

Reply:
let's change it to create.

Reply:
maybe `Failed to "construct"..`?

Reply:
Done !